### PR TITLE
Sprint 52

### PR DESCRIPTION
**Pull request by ericsorenson**
# Sprint 52 

### Closed Issues 

- [6909 BUG: formattedJudgeName will fail](https://github.com/flexion/ef-cms/issues/6909) 
- [6959 Security Risk: isSealed](https://github.com/flexion/ef-cms/issues/6959) 
- [6972 BUG: Eligible Case list : L & P cases are not being sorted to top ](https://github.com/flexion/ef-cms/issues/6972) 

### Open Issues 

- [4661 Docket Clerk: Notice of Change of Address/Phone #/Both ?](https://github.com/flexion/ef-cms/issues/4661) 
- [6684 Pending Report: Loading Issue](https://github.com/flexion/ef-cms/issues/6684) 
- [6836 Performance Tuning: Search Results (casePublicSearch, caseAdvancedSearch, advancedDocumentSearch & getPractitionersByName)](https://github.com/flexion/ef-cms/issues/6836) 
- [6961 Security Risk: isStricken ](https://github.com/flexion/ef-cms/issues/6961) 
- [6978 BUG: Edit link displaying for Closed Trial Session](https://github.com/flexion/ef-cms/issues/6978) 
- [6780 BUG: (Mobile) Mobile is still showing “Need help selecting a document” link on the “What Document Are you Filing?](https://github.com/flexion/ef-cms/issues/6780)

### CODE REVIEW

File: .nova/Configuration.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (julialeague) [inline .nova/Configuration.json]: should this file be .gitignored?

File: shared/src/business/entities/DocketEntry.js [hunk 2]
```diff
@@ -210,7 +211,14 @@ and contact info from the case detail
  * @param {object} caseDetail the case detail
  */
 DocketEntry.prototype.generateFiledBy = function (caseDetail) {
-  if (!this.filedBy) {
+  const isNoticeOfContactChange = NOTICE_OF_CHANGE_CONTACT_INFORMATION_EVENT_CODES.includes(
+    this.eventCode,
+  );
+
+  const shouldGenerateFiledBy =
+    !this.filedBy && !(isNoticeOfContactChange && this.isAutoGenerated);
```
Comment (julialeague) [inline shared/src/business/entities/DocketEntry.js:219]: Could we add tests for this new logic?

File: shared/src/business/entities/DocketEntry.test.js [hunk 3]
```diff
@@ -1943,4 +1944,50 @@ describe('DocketEntry entity', () => {
       expect(docketEntry.isValid()).toBeTruthy();
     });
   });
+
+  describe('generateFiledBy', () => {
```
Comment (julialeague) [inline shared/src/business/entities/DocketEntry.test.js]: There's already a describe block in this test file that's called "generate filed by string" testing other filedBy  scenarios - it seems like these new tests could be organized along with those.

File: shared/src/business/useCases/getDownloadPolicyUrlInteractor.js [hunk 4]
```diff
@@ -147,6 +147,10 @@ exports.getDownloadPolicyUrlInteractor = async ({
           throw new UnauthorizedError(
             'Unauthorized to view document at this time.',
           );
+        } else if (docketEntryEntity.isStricken) {
```
Comment (julialeague) [inline shared/src/business/useCases/getDownloadPolicyUrlInteractor.js:150]: Can we get a test for this?

File: web-client/integration-tests-public/journey/unauthedUserAttemptsToViewStrickenDocumentUnsuccessfully.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+export const unauthedUserAttemptsToViewStrickenDocumentUnsuccessfully = test => {
+  return it('View stricken document unsuccessfully', async () => {
+    await expect(
+      test.runSequence('openCaseDocumentDownloadUrlSequence', {
+        docketEntryId: test.docketEntryId,
+        docketNumber: test.docketNumber,
+        isPublic: true,
```
Comment (julialeague) [inline web-client/integration-tests-public/journey/unauthedUserAttemptsToViewStrickenDocumentUnsuccessfully.js:7]: It's stricken but it's public? Is that right?

Comment (julialeague) [reply web-client/integration-tests-public/journey/unauthedUserAttemptsToViewStrickenDocumentUnsuccessfully.js:7]: Never mind, I think this is just saying the searcher is not an authenticated user. 👍

**Comment by julialeague** [comment]
- [x] review the pull request itself, to get oriented
	- [x] read the description of the pull request, which should summarize the changes made
	- [x] read through every task on the Scrum board that's encompassed by this pull request
	- [x] read the description of the commits that comprise the pull request
- [x] If changes were made in the UI (if not, skip):
  - [x] If desired, run `docker system prune` to remove any unused docker images from previous code review.
  - [x] Fetch the pull request for the sprint (e.g., `git fetch origin pull/{PR #}/head:sprint-{Sprint #}`), and then switch to that branch (e.g. `git checkout sprint-{Sprint #}`)
  - [x] stand up the site locally, with `./docker-run.sh`
	- [x] test all functionality in all major browsers, emphasizing the functionality that this pull request addresses
	- [x] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
	- [x] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/)
	- [x] test in Mobile Safari and Mobile Chrome (or an emulator like Chrome DevTools), with the caveat that not all internal Court functionality will be necessary on these platforms
	- [x] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [x] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [x] review for accessibility
		- [x] use an automated audit tool, such as Chrome Audit or aXe
		- [x] navigate site only with the keyboard
		- [x] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [x] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [x] [review static code analysis results](https://sonarcloud.io/organizations/flexion-github/projects) in the vendor’s SonarCloud environment
- [x] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [x] skim all new code, in the context of existing code, [looking for problems](#what-we-look-for) (knowing that the vast majority of new code will be covered by tests)
- [x] review all tests
  - [x] look at code coverage of tests in [the vendor's SonarCloud environment](https://sonarcloud.io/organizations/flexion-github/projects)
  - [x] methodically review all new tests for correctness, quality of naming
- [x] determine what code isn’t tested, review that rigorously
- [x] review documentation to ensure that it matches changes
  - [x] user-visible changes (including API users like the IRS) are documented in CHANGELOG.md (which is linked from US Tax Court website).
- [x] provide comments on the pull request on GitHub, as necessary
  - [x] for comments that are specific to a particular line of code, comment on those specific lines
- [x] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

File: .nova/Configuration.json (hunk 1 above)
Comment (ericsorenson) [reply .nova/Configuration.json]: That was an accidental commit by me, I'll remove it!

File: shared/src/business/entities/DocketEntry.test.js (hunk 3 above)
Comment (ericsorenson) [reply shared/src/business/entities/DocketEntry.test.js]: Good call, update coming!

File: .nova/Configuration.json (hunk 1 above)
Comment (ericsorenson) [reply .nova/Configuration.json]: Removed!

File: shared/src/business/entities/DocketEntry.test.js (hunk 3 above)
Comment (ericsorenson) [reply shared/src/business/entities/DocketEntry.test.js]: Updated!

File: shared/src/business/useCases/getDownloadPolicyUrlInteractor.js (hunk 4 above)
Comment (ericsorenson) [reply shared/src/business/useCases/getDownloadPolicyUrlInteractor.js:150]: Test added!

**Comment by julialeague** [comment]
@mmarcotte, I recommend you accept this PR into the court's environments. Thank you!

**Review by julialeague — APPROVED** [review]

**Pull request merged**